src/main/java/junmin/netflix/costreportservice/service/EPCostReportService.java

```diff
@@ -0,0 +1,169 @@
+package junmin.netflix.costreportservice.service;
+
+import junmin.netflix.costreportservice.exception.ErrorEnum;
+import junmin.netflix.costreportservice.exception.InvalidProductionException;
+import junmin.netflix.costreportservice.exception.InvalidReportException;
+//import junmin.netflix.costreportservice.model.EPItemizedCostEntity;
+import junmin.netflix.costreportservice.model.EPTotalCostEntity;
+import junmin.netflix.costreportservice.pojo.EPCostRecord;
+import junmin.netflix.costreportservice.repository.EPItemizedCostRepository;
+import junmin.netflix.costreportservice.repository.EPTotalCostRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+
+@Service
+public class EPCostReportService {
+	private EPTotalCostRepository epTotalCostRepo;
+	private EPItemizedCostRepository epItemizedCostRepo;
+
+	public EPCostReportService(EPTotalCostRepository epTotalCostRepo, EPItemizedCostRepository epItemizedCostRepo){
+		this.epTotalCostRepo = epTotalCostRepo;
+		this.epItemizedCostRepo = epItemizedCostRepo;
+	}
+
+	private final Logger LOGGER = LoggerFactory.getLogger(EPCostReportService.class);
+	private final String AMORIZEDCODE = "001";
+
+
+	public List<EPCostRecord> processEPCostReport(String prodName, List<EPCostRecord> costs) throws InvalidReportException
+	{
+		validateProdName(prodName);
+		String epCode = getEPCodeFromEPReport(costs);
```

Comment:
sure, copy same reply from above: 
Based on the requirement doc, it asked three new endpoints to process single EP cost report, production cost report and production cost report with amortized amount. So that what I followed. Imagine for process single EP cost report, if you have two EP code in the report, I would consider it to be error and give alert.
If we do not require 3 new endpoints, I would just combine all three scenarios into one POST operation: production cost report with amortized amount, it's sufficient to cover all cases.